src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/Cmdlets/GetAzureRmVMGuestPolicyReport.cs

```diff
@@ -0,0 +1,180 @@
+    using Microsoft.Azure.Management.GuestConfiguration.Models;
+
+    /// <summary>
+    /// Gets InGuest/Vm Policy reports
+    /// </summary>
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VMGuestPolicyReport", DefaultParameterSetName = ParameterSetNames.VmNameScope), 
+        OutputType(typeof(IList<GuestConfigurationAssignment>), typeof(IList<GuestConfigurationAssignmentReport>))]
+    public class GetAzureRmVMGuestPolicyReport : GuestConfigurationCmdletBase
+    {
+        [Parameter(ParameterSetName = ParameterSetNames.VmNameScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = ParameterHelpMessages.ResourceGroupName)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = ParameterHelpMessages.ResourceGroupName)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = ParameterHelpMessages.ResourceGroupName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.VmNameScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = ParameterHelpMessages.VMName)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = ParameterHelpMessages.VMName)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = ParameterHelpMessages.VMName)]
+        [ValidateNotNullOrEmpty]
+        public string VMName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.InitiativeName)]
+        [ValidateNotNullOrEmpty]
+        public string InitiativeName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.InitiativeId)]
+        [ValidateNotNullOrEmpty]
+        public string InitiativeId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ReportIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.ReportId)]
+        [ValidateNotNullOrEmpty]
+        public string ReportId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
```

Comment:
Optional parameters that are in all parameter sets don't need to mention parameters et name; by default parameters with no parameter set name are in all parameter sets

Reply:
Please do not make optional parameters positional

Reply:
Position removed for optional parameters